### PR TITLE
refactor(amazon/image): Convert amazon image reader to a TS class

### PR DESCRIPTION
**Pull request by christopherthielen**

### CODE REVIEW

File: app/scripts/modules/amazon/src/image/image.reader.ts [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+  imageName: string;
+  tags: {
+    [tag: string]: string;
+  };
+  tagsByImageId: {
+    [imageId: string]: IAmazonImage['tags'];
+  };
+}
+
+export class AwsImageReader {
```
Comment (jrsquared) [inline app/scripts/modules/amazon/src/image/image.reader.ts]: Looks like this doesn't need angular at all... how about just make it a static non-angular class while you're here?

Comment (christopherthielen) [reply app/scripts/modules/amazon/src/image/image.reader.ts]: fine

Comment (christopherthielen) [reply app/scripts/modules/amazon/src/image/image.reader.ts]: nope. this is still needed by aws module 😢 

https://github.com/spinnaker/deck/blob/e1a33930a6b522f9349c4ccbe1abc4aa93bb6e1f/app/scripts/modules/amazon/src/aws.module.ts#L82-L92

Comment (christopherthielen) [reply app/scripts/modules/amazon/src/image/image.reader.ts]: De-angularized this in a new commit.

**Pull request merged**